impacted-test-engine/src/main/java/com/teamscale/test_impacted/engine/executor/ImpactedTestsExecutor.java

```diff
@@ -0,0 +1,92 @@
+package com.teamscale.test_impacted.engine.executor;
+
+import com.teamscale.client.ClusteredTestDetails;
+import com.teamscale.client.CommitDescriptor;
+import com.teamscale.client.TeamscaleClient;
+import com.teamscale.client.TestClusterForPrioritization;
+import com.teamscale.report.testwise.model.TestExecution;
+import com.teamscale.test_impacted.controllers.ITestwiseCoverageAgentApi;
+import com.teamscale.test_impacted.engine.options.ServerOptions;
+import com.teamscale.test_impacted.test_descriptor.TestDescriptorUtils;
+import org.junit.platform.commons.logging.Logger;
+import org.junit.platform.commons.logging.LoggerFactory;
+import org.junit.platform.engine.TestDescriptor;
+import org.junit.platform.engine.UniqueId;
+import retrofit2.Response;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/** Test executor that only executes impacted tests and collects test wise coverage for the executed tests. */
+public class ImpactedTestsExecutor extends TestWiseCoverageCollectingTestExecutor {
```

Comment:
In TS code we consistently use the casing `testwise` / `TestwiseCoverage` not sure what's more correct, but we should keep it consistent.